### PR TITLE
validate num repos between creds

### PR DESCRIPTION
**Pull request by thingscouldbeworse**

### CODE REVIEW

File: jf_agent/main.py [hunk 1]
```diff
@@ -132,6 +132,29 @@ def main():
...
+                'validate_num_repos' in jellyfish_endpoint_info.jf_options.keys() and
+                jellyfish_endpoint_info.jf_options['validate_num_repos']
+            ):
+                validate_num_repos_go_for = True
+            else:
+                validate_num_repos_go_for = False
+        except Exception as e:
+            agent_logging.log_and_print(logger, logging.WARNING, msg=f"Could not get `jf_options` instead got {e}."
+                                                                     f"not critical, continuing.")
+            validate_num_repos_go_for = False
```
Comment (gavinpitt-jf) [inline jf_agent/main.py:147]: `jellyfish_endpoint_info` is a dict, right? I think we can get rid of this try catch block if we use the `.get()` call above

Comment (gavinpitt-jf) [reply jf_agent/main.py:147]: Oh whoops, disregard my comments. Didn't see it was a `namedtuple`

File: jf_agent/main.py [hunk 2]
```diff
@@ -132,6 +132,29 @@ def main():
...
         jellyfish_endpoint_info = obtain_jellyfish_endpoint_info(config, creds)
+        try:
+            if (
+                jellyfish_endpoint_info.jf_options and
+                'validate_num_repos' in jellyfish_endpoint_info.jf_options.keys() and
+                jellyfish_endpoint_info.jf_options['validate_num_repos']
+            ):
+                validate_num_repos_go_for = True
+            else:
+                validate_num_repos_go_for = False
```
[the remark this answers is not in the document]
Comment (gavinpitt-jf) [reply jf_agent/main.py:143]: Didn't realize jf_options was a `namedtuple`... I still think you could leverage `.get()` on the dict to simplify the if statement

[the remark this answers is not in the document]
Comment (thingscouldbeworse) [reply jf_agent/main.py]: yeah agreed, here's that: https://github.com/Jellyfish-AI/jf_agent/pull/242/commits/ed01ef87ff36bc22ae2a5a2d91082ba9c9f86c99

**Pull request closed without merging**